phpunit/class-wp-rest-block-patterns-controller-test.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+class WP_REST_Block_Patterns_Controller_Test extends WP_Test_REST_Controller_Testcase {
+	protected static $admin_id;
+
+	public function set_up() {
+		parent::set_up();
+		switch_theme( 'emptytheme' );
+	}
+
+	public static function wpSetupBeforeClass( $factory ) {
+		self::$admin_id = $factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+	}
+
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+		$this->assertArrayHasKey(
+			'/__experimental/block-patterns/patterns',
+			$routes,
+			'The patterns route does not exist'
+		);
+	}
+
+	public function test_get_items() {
+		// Change this when the registered Core patterns change.
+		$expected_names  = array(
```

Comment:
Should we make this more resilient by probably unregistering existing patterns, register some test patterns and then restore?

Reply:
Yes, I'll try to do that, the tests are too fragile in the current form. I'll probably have to create a mock version of the two registries.